src/sparseml/transformers/utils/sparse_model.py

```diff
@@ -26,22 +26,28 @@
     AutoModelForQuestionAnswering,
     AutoModelForSequenceClassification,
     AutoModelForTokenClassification,
+    PreTrainedModel,
 )
 from transformers.file_utils import WEIGHTS_NAME
 
 from sparseml.pytorch.model_load.helpers import (
     apply_recipe_structure_to_model,
     log_model_load,
 )
+from sparseml.transformers.compression.utils import (
+    add_save_compressed_method,
+    infer_compressor_from_model_config,
+    modify_save_pretrained,
+)
 from sparseml.transformers.utils.helpers import resolve_recipe
 from sparseml.utils import download_zoo_training_dir
 from sparseml.utils.fsdp.context import main_process_first_context
 
 
-__all__ = ["SparseAutoModel", "SparseAutoModelForCausalLM", "get_shared_tokenizer_src"]
+_LOGGER = logging.getLogger(__name__)
 
 
-_LOGGER = logging.getLogger(__name__)
+__all__ = ["SparseAutoModel", "SparseAutoModelForCausalLM", "get_shared_tokenizer_src"]
 
 
 class SparseAutoModelForCausalLM(AutoModelForCausalLM):
```

Comment:
Do you think this should be implemented in `SparseAutoModelForCausalLM` or `SparseAutoModel`?

Just curious since if we expand to encoders / encoder-decoder / multi-modal, then all of this will have to be ported

Maybe something we do in a future PR

Reply:
`SparseAutoModel` has a lot of legacy code right now for old transformers flows, I do agree it makes sense to put this code somewhere more general but I think we should hold off until we're actually adding in additional model types

Reply:
This makes sense